drivers/scd30/scd30.c

```diff
@@ -0,0 +1,396 @@
+{
+    int ret = 0;
+
+    // Check if input is valid
+    if (!_is_valid_interval(interval)) {
+        DEBUG("[scd30] Interval value out of range\n");
+        return SCD30_INVALID_VALUE;
+    }
+    if (!_is_valid_press_comp(apc)) {
+        DEBUG("[scd30] Ambient pressure compensation value out of range\n");
+        return SCD30_INVALID_VALUE;
+    }
+
+    ret = scd30_set_param(dev, SCD30_INTERVAL, *interval);
+    if (ret != 0) {
+        return ret;
+    }
+//    xtimer_usleep(SCD30_READ_WRITE_SLEEP_US);
+    ret = scd30_set_param(dev, SCD30_START, *apc);
+    if (ret != 0) {
+        return ret;
+    }
+    return SCD30_OK;
+}
+
+int8_t scd30_stop_measurements(scd30_t *dev)
+{
+    uint8_t buffer[2];
+    uint16_t cmd = SCD30_STOP;
+    int ret;
+
+    buffer[0] = (uint8_t)(cmd >> 8);
+    buffer[1] = (uint8_t)cmd;
+    i2c_acquire(SCD30_I2C);
+    ret = i2c_write_bytes(SCD30_I2C, SCD30_I2C_ADDRESS, buffer, 2, 0);
+    i2c_release(SCD30_I2C);
+    if (ret != 0) {
+        DEBUG("[scd30]: Error stopping measurements.\n");
+    }
+    return SCD30_OK;
+}
+
+/**
+ *      intern function to read data
+ *
+ *      @param  dev      device
+ *		@param  result	 struct in which data read is stored
+ *      @return          SCD30_OK on success
+ */
+static int8_t _scd30_read_data(scd30_t *dev, scd30_measurement_t *result)
+{
+    uint8_t buffer[18];
+    int ret = 0;
+    uint16_t cmd = SCD30_DATA;
+
+    buffer[0] = (uint8_t)(cmd >> 8);
+    buffer[1] = (uint8_t)cmd;
+    i2c_acquire(SCD30_I2C);
+    ret = i2c_write_bytes(SCD30_I2C, SCD30_I2C_ADDRESS, buffer, 2, 0);
+    if (ret != 0) {
+        i2c_release(SCD30_I2C);
+        return ret;
+    }
+    ret = i2c_read_bytes(SCD30_I2C, SCD30_I2C_ADDRESS, buffer, 18, 0);
+    if (ret != 0) {
+        DEBUG("[scd30] read_data: ret %i.\n", ret);
+        i2c_release(SCD30_I2C);
+        return ret;
+    }
+    i2c_release(SCD30_I2C);
+
+    if (_scd30_crc_check(buffer, 2, buffer[2]) != 0) {
+        DEBUG("1\n");
+        return SCD30_CRC_ERROR;
+    }
+    if (_scd30_crc_check(buffer + 3, 2, buffer[5]) != 0) {
+        return SCD30_CRC_ERROR;
+    }
+    if (_scd30_crc_check(buffer + 6, 2, buffer[8]) != 0) {
+        return SCD30_CRC_ERROR;
+    }
+    if (_scd30_crc_check(buffer + 9, 2, buffer[11]) != 0) {
+        return SCD30_CRC_ERROR;
+    }
+    if (_scd30_crc_check(buffer + 12, 2, buffer[14]) != 0) {
+        return SCD30_CRC_ERROR;
+    }
+    if (_scd30_crc_check(buffer + 15, 2, buffer[17]) != 0) {
+        return SCD30_CRC_ERROR;
+    }
+
+    result->co2_concentration = _raw_val_to_uint32((uint32_t)(
+                                                       (((uint32_t)buffer[0]) <<
+    24) |
+                                                       (((uint32_t)buffer[1]) <<
+    16) |
+                                                       (((uint32_t)buffer[3]) <<
+    8) |
+                                                       ((uint32_t)buffer[4])));
```

Comment:
```suggestion
    result->co2_concentration = byteorder_bebuftohl(&buffer[0]);
```

You will have to rebase though as that function is rather new.
Also, why multiply by 100?

Reply:
multiply by 100 is to store two points of decimal accuracy without using float

Reply:
But you *are* using float

Reply:
Yeah the sensor gives the float value in its IEEE754 representation. But from our last discussion in #14329, I reduced the dependency on float as much as possible

Reply:
We should avoid using float values if possible, but if the sensor gives us float values than we should just use them :wink: 
At that point we have to pull in the soft-float library on non-FPU MCUs anyway, so we might as well use it.
If we get integer values from a sensor we shouldn't convert those to float as it will introduce imprecision and overhead, but if float is what the sensor gives us, it won't get any more precise by converting it to int.

Since this is a rather unusual property, please add a comment about it in the code.

Reply:
Oh yeah, makes sense